packages/uikit/src/widgets/Modal/ModalContext.tsx

```diff
@@ -105,14 +106,18 @@ const ModalProvider: React.FC<React.PropsWithChildren> = ({ children }) => {
 
   const handleOverlayDismiss = useCallback(() => {
     if (closeOnOverlayClick) {
+      const customOnDismiss = get(modalNode, "props.customOnDismiss") as any;
+      customOnDismiss?.();
       handleDismiss();
     }
-  }, [closeOnOverlayClick, handleDismiss]);
+  }, [closeOnOverlayClick, handleDismiss, modalNode]);
 
   const providerValue = useMemo(() => {
     return { isOpen, nodeId, modalNode, setModalNode, onPresent: handlePresent, onDismiss: handleDismiss };
   }, [isOpen, nodeId, modalNode, setModalNode, handlePresent, handleDismiss]);
 
+  const handleAnimationStart = useCallback(() => animationHandler(animationRef.current), [animationRef]);
```

Comment:
No need to put into dependency array?
```suggestion
  const handleAnimationStart = useCallback(() => animationHandler(animationRef.current), []);
```